src/Layers/xrRenderGL/glSH_Texture.cpp

```diff
@@ -208,13 +208,6 @@ void CTexture::Load()
 
             pSurface = pTexture;
             desc = GL_TEXTURE_2D;
-            GLenum err = glGetError();
-            if (err != GL_NO_ERROR)
-            {
-                FATAL_F("Invalid video stream: 0x%x", err);
-                xr_delete(pTheora);
-                pSurface = 0;
-            }
```

Comment:
Maybe it's better to do like this?
```cpp
            GLenum err = glGetError();
            if (err != GL_NO_ERROR)
            {
                Msg("! Invalid video stream: 0x%x", err);
                // Sometimes it crashes if video stream is correct
                // XXX: Fix!
                //FATAL_F("Invalid video stream: 0x%x", err);
                //xr_delete(pTheora);
                //pSurface = 0;
            }
```

Reply:
Not. Error OpenGL appears earlier, and is not related in any way with the "video stream". Just a error or a few errors remained in the stack of errors, and one of them got here. Where an error (s) appeared, it is difficult to determine with the help of glGetError (), since it needs to be called after each gl * () call to know for sure. Thus is more broad Msg("! Some erroneous use usage of OpenGL detected"); is good.

Reply:
More right way is writing error handler with ability skip uncritical errors.